File: sample.py
import os
import json
import requests
import base64

#globals
vrate_api = "https://api2.vrate.net/nudityrating"
vrate_headers = {"X-Auth":"<enter authorization key, contact info@vrate.net",'Content-Type':'application/json'}

if __name__ == "__main__":
    #***** Analyze an image on a public url and print result ******
    image_url = "http://30.media.tumblr.com/tumblr_m20ljbf2wy1r3e62yo1_500.jpg"
    image = requests.get(image_url).content #download image
    encoded_string = base64.b64encode(image) #convert to base64
    vrate_payload = json.dumps({"payload":encoded_string}) #create the input payload for vRate
    response = json.loads(requests.post(vrate_api,headers=vrate_headers,data=vrate_payload).content) #call api & get rating
    print response


    #***** Analyze a local image and print result *****************
    file_img = 'sample.jpg'
    with open(file_img, "rb") as image_file:
        encoded_string = base64.b64encode(image_file.read())
    vrate_payload = json.dumps({"payload":encoded_string}) #create the input payload for vRate
    response = json.loads(requests.post(vrate_api,headers=vrate_headers,data=vrate_payload).content) #call api & get rating
    print response

File: sample.js
//globals
var request = require('request');

// ------url input
var USER_DATA = {
    'payload':'http://30.media.tumblr.com/tumblr_m20ljbf2wy1r3e62yo1_500.jpg'
}


// ----- local file input -  we need to send the input as base 64 conversion for local file.
var local_file = 'sample.jpg'  // provide your local file path
var buffer = fs.readFileSync(local_file)
console.log("response" + response);
console.log("body" + buffer.toString("base64"));

var USER_DATA = {
    'payload': buffer.toString("base64")
}

var options = {
    method: 'POST',
    url: 'https://api2.vrate.net/nudityrating"',
    headers: {
       'Content-Type': 'application/json',
        "x-api-key":"UhS51aGn0K34JcGWiWnJ3azcFS8hAdiV8gjdv7wH"
    },
    json: USER_DATA

};

// Analyze a INPUT image and print the result

function callback(error, response, body) {
    if (!error) {
        var info = JSON.parse(JSON.stringify(body));
        console.log(info);
 	console.log("Rating code: " + info.RatingCode);
 	console.log("Rating Description: " + info.RatingDescription);
    }
    else {
        console.log('Error happened: '+ error);
    }
//send request to vRate API
request(options, callback);
